internal/storage/image.go

```diff
@@ -398,6 +404,56 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, imageName string) (bool, error) {
+	log.Debugf(ctx, "Checking if image %s is allowed to run", imageName)
+
+	ref, err := svc.getRef(imageName)
+	if err != nil {
+		return false, fmt.Errorf("get ref for name or ID: %w", err)
+	}
+
+	// Check the cache first
+	var policyContext *signature.PolicyContext
+	item := svc.policyContexts.Get(imageName)
```

Comment:
This needs to be keyed on `systemContext` — or probably directly on the namespace name CRI-O uses to decide on which policy to look for.

Otherwise two different namespaces with two different policies, but pulling the same `imageName`, would interfere.

Reply:
Ah makes sense, I'm using `systemContext.SignaturePolicyPath` now, which should be implicitly refer to the namespace.